CIP-0PRS/CIP-0PRS.lagda.md

```diff
@@ -0,0 +1,1355 @@
+CIP: ?
+Category: Consensus
+Status: Proposed
+Authors:
+  - Arnaud Bailly <arnaud.bailly@iohk.io>
+  - Brian W. Bush <brian.bush@iohk.io>
+  - Yves Hauser <yves.hauser@iohk.io>
+  - Hans Lahe <hans.lahe@iohk.io>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/?
+Created: 2024-07-31
+License: Apache-2.0
+---
+
+> [!IMPORTANT]
+>
+> Decisions needed:
+> - [ ] Do we want the CIP to be type-checked by the Agda compiler?
+>     - [ ] *Option 1:* Write the CIP as a single, self-contained `.lagda.md` file.
+>         - *Pro:* completely self-contained
+>         - *Con:* need to flatten our `src/` files into a single file
+>     - [ ] *Option 2:* Put all of the Agda code in a `src/` folder within the CIP's folder.
+>         - *Pro:* the existing `src/` can be copied over into the CIP, after a little clean-up
+>         - *Con:* the Agda specification won't appear in the top-level CIP file, but will be reachable by links
+>     - [ ] *Option 3:* Copy-and-paste the Agda code from our `src/` into the CIP
+>         - *Pro:* smaller CIP file
+>         - *Con:* not type-checked
+> - [ ] Should we only include the ALBA certificates in the specification? It's confusing to propose two types of certificate. Maybe the Mithril-certificate section could be moved to an appendix?
+> - [ ] The benchmarks don't properly belong in the "Specification" section, so they're being moved to "Resources".
+> - [ ] How much of the evidence and analyses should be embedded in this document versus referencing the technical reports?
+> - [ ] Editing
+>     - [ ] Edit in a folder of `peras-design`
+>     - [ ] Edit in a fork of the CIP repo
+> - [ ] Include the research team as CIP authors?
+> - [ ] In order to speed things up, should we also draft the meta-CIP that creates the "Consensus" category?
+
+> [!NOTE]
+> A "Consensus" category must be created by a new CIP before this CIP can be submitted within that category.
+>
+> Existing categories:
+> 
+> - Meta     | For meta-CIPs which typically serves another category or group of categories.
+> - Wallets  | For standardisation across wallets (hardware, full-node or light).
+> - Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+> - Metadata | For proposals around metadata (on-chain or off-chain).
+> - Tools    | A broad category for ecosystem tools not falling into any other category.
+> - Plutus   | Changes or additions to Plutus
+> - Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+> - Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+* Look at [Leios spec](https://github.com/cardano-foundation/CIPs/blob/ouroboros-leios/CIP-0079/README.md) for inspiration
+* [CIP-001](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0001) guidelines
+
+## Abstract
+<!-- A short (\~200 word) description of the proposed solution and the technical issue being addressed. -->
```

Comment:
```suggestion
<!-- A short (\~200 word) description of the proposed solution and the technical issue being addressed. -->
```
Now that the content is in place here you can (and should, before editing really begins) remove the comment scaffolding from the template: both here & in other places in the document.